src/reconcile/rec_write.c

```diff
@@ -187,35 +193,35 @@ typedef struct {
 		 * column-store key.
 		 */
 		WT_ITEM key;		/* Promoted row-store key */
+
+		/*
+		 * For each boundary that we create at the split percentage, we
+		 * also maintain details of where a boundary would be, if we
+		 * split at the minimum split percentage.
+		 * While reconciling pages, at any given time, we maintain two
+		 * split chunks in the memory to be written out as pages. As we
+		 * get to the last two chunks, if the last one turns out to be
+		 * smaller than the minimum split size, we go back into the
+		 * penultimate chunk and split at this minimum split size
+		 * boundary. This moves some data from the penultimate chunk to
+		 * the last chunk, hence increasing the size of the last page
+		 * written without decreasing the penultimate page size beyond
+		 * the minimum split size.
+		 */
+		uint32_t min_bnd_entries; /* No of entries at min split pct */
+		/* Next chunk's starting offset at min split pct */
+		size_t min_bnd_offset;
+		/* Next chunk's starting record number at min split pct */
+		uint64_t min_bnd_recno;
+		/* Next chunk's starting key at min split pct */
+		WT_ITEM min_bnd_key;
```

Comment:
I like the idea of renaming boundary information as max/min. Though min suggests a global minimum of 50% (the one user can set and the one we will readjust to if the last chunk is not large enough), max will be 50 to 100 %, equal to the split percentage user has configured. Max doesn't signify the maximum split percentage that can be set (100%). If "max" doesn't create a confusion like that, I am happy to rename.

You are right, we do not need min_XXX for all the boundaries. Initially I thought of and discussed that approach, and then decided to have this information for each boundary. With respect to performance, I don't think there will be any difference and I suppose extra information per boundary doesn't consume too much memory either. It looked simpler and cleaner to track per boundary. Having only one set of min_XXX fields requires me to have an additional bool flag to determine the state of this information for the current boundary. At the moment I initialise min_bnd_offset to 0, and can check if I have crossed the minimum by checking if it is still 0, not possible without extra field or more complex calculations on the offset and current size if maintaining one set of min_XXX fields.

If you don't have strong reasons against tracking information for all the boundaries, I am inclined to keep it.